### PR TITLE
Simple converson tool

### PR DESCRIPTION
**Pull request by ElTitoDG**
Added the conversion tool to the right of the speed slider. It consists of 4 fields for each format: Decimal, Hex, Binary and Ascii, and is automatically updated when the user types in any of the fields.

Respond to issue #25

I am not sure if the way it is integrated in the code is the correct one, in reference to how the project is structured, but I thought it would work if it is inside the tools folder.

### CODE REVIEW

File: src/rars/tools/ConversionTool.java [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+        JLabel hexLabel = new JLabel("Hex:");
+        JTextField hexField = new JTextField(7);
+        hexField.setToolTipText("Enter a hexadecimal number with or without '0x' prefix (e.g., 0x61)");
+
+        JLabel binaryLabel = new JLabel("Binary:");
+        JTextField binaryField = new JTextField(10);
+        binaryField.setToolTipText("Enter a binary number with or without '0b' prefix (e.g., 0b01100001)");
+
+        JLabel charLabel = new JLabel("Ascii:");
```
Comment (privat) [inline src/rars/tools/ConversionTool.java:34]: ASCII (uppercased)

File: src/rars/tools/ConversionTool.java [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    public JPanel createConversionToolPanel() {
+        JPanel panel = new JPanel();
+        panel.setLayout(new FlowLayout());
+
+        // Create labels and text fields for each format
+        JLabel decimalLabel = new JLabel("Decimal:");
+        JTextField decimalField = new JTextField(5);
+        decimalField.setToolTipText("Enter a decimal number (e.g., 97)");
+
+        JLabel hexLabel = new JLabel("Hex:");
```
Comment (privat) [inline src/rars/tools/ConversionTool.java:26]: I would prefer something more consistent in the labels. Either Decimal/Hexadecimal/Binary or Dec/Hex/Bin

File: src/rars/tools/ConversionTool.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+     * @param charField         The character text field.
+     */
+    private void updateFromDecimal(JTextField decimalField, JTextField hexField, JTextField binaryField, JTextField charField) {
+        if (updating) return;
+        updating = true;
+        try {
+            int decimal = Integer.parseInt(decimalField.getText());
+            hexField.setText("0x" + Integer.toHexString(decimal));
+            binaryField.setText("0b" + Integer.toBinaryString(decimal));
+            charField.setText(Character.toString((char) decimal));
```
Comment (privat) [inline src/rars/tools/ConversionTool.java:147]: Maybe just write "" if the number > 127 ; above 127 it's not strictly ASCII. Moreover, by casting to char this wrap for number > 65535 (so it's not strictly Unicode either)

**Review by privat — COMMENTED** [review]
This is great. I have some nitpicks about the label but good job

**Comment by ElTitoDG** [comment]
I've added the ascii limit, but I don't understand the Unicode limit because the conversion only goes for simply ascii

**Comment by privat** [comment]
> I've added the ascii limit, but I don't understand the Unicode limit because the conversion only goes for simply ascii

`char` in java is 2 bytes, so goes from 0 to 0xFFFF (65535). Unicode goes up to 0x10FFFF

**Pull request merged**